### PR TITLE
[WIP] [SYSTEMML-445] Upgraded CUDA/CuDNN versions and added LSTM, batch normalization kernels

### PR DESCRIPTION
**Pull request by niketanpansare**
- Upgraded the CUDA and CuDNN version.
- Added GPU usage documentation.
- Clean up GPU memory manager.
- Improved the performance of NN batch normalization layer by avoiding unnecessary computation.
- Added `lstm`, `batch_norm2d` and `batch_norm2d_backward` builtin functions with their respective GPU implementation. The CPU implementation will be added in a separate PR.

@nakul02 can you please review this PR ?

@bertholdreinwald @mboehm7 @prithvirajsen

### CODE REVIEW

**Comment by akchinSTC** [comment]
Can an Admin verify this patch?

**Comment by niketanpansare** [comment]
I will try to split this PR into smaller commits

**Comment by nakul02** [comment]
I haven't taken a look at this, but maybe you could wait till jcuda 9.1 or 9.2 is out and then modify this PR accordingly?
Take a look at https://github.com/jcuda/jcuda-main/issues/25

**Comment by niketanpansare** [comment]
I wanted to break this PR into smaller commits and have upgraded to CUDA 9 in the commit: https://github.com/apache/systemml/commit/72fd8fda39062c9059e60d054f263ea1d92bf517 ... We can switch to 9.1 or 9.2 when it is released. Upgrading to CUDA 9 is blocker for many kernels: lstm, batch norm (forward and backward).

**Comment by nakul02** [comment]
Ok, let us upgrade to jcuda >= 9.1 after contents of this PR are merged in.
